modules/iap/iap.tf

```diff
@@ -21,6 +21,23 @@ data "google_project" "project" {
   project_id = var.project_id
 }
 
+## Check if domain and members_allowlist is provided
+data "null_data_source" "domain_validation" {
+  inputs = {}
+
+  lifecycle {
+    precondition {
+      condition     = var.domain != ""
+      error_message = "IAP configuration requires domain name, Please provide a valid domain name for ${var.app_name} application."
+    }
+
+    precondition {
+      condition     = length(var.members_allowlist) != 0
```

Comment:
member allowlist can be empty, they have the right to add them later

Reply:
I think this just makes the experience more confusing, most users would want to allowlist at least themselves - they'll leave this blank and wonder why they can't login later. Prefer to fail early and force them to add something to the allowlist.